learn-typescript-linting.md

```diff
@@ -0,0 +1,407 @@
+
+```bash
+npm run compile
+npm run start
+```
+
+It is recommended you do not modify the `.ts` files as they are specifically set up to show off unique aspects of linting typescript projects.
+
+## 2 Adding ESLint
+[`[toc]`](#table-of-contents)
+
+Install `eslint`, `@typescript-eslint/eslint-plugin`, `@typescript-eslint/parser` as developer dependencies. Initialize an empty eslint configuration file. I prefer to use `.json` for configuration files.
+
+```bash
+npm i -D eslint @typescript-eslint/{eslint-plugin,parser}
+touch .eslintrc.json
+```
+
+### 2.1 Initializing .eslintrc
+[`[toc]`](#table-of-contents)
+
+Add the following to `.eslintrc.json`
+```json
+{
+  "extends": ["eslint:recommended", "plugin:@typescript-eslint/recommended"],
+  "parser": "@typescript-eslint/parser",
+  "plugins": ["@typescript-eslint"],
+  "env": { "node": true },
+  "parserOptions": {
+    "ecmaVersion": 5,
+    "sourceType": "module"
+  }
+}
+```
+
+### 2.2 Specifying environments
+[`[toc]`](#table-of-contents)
+
+The `env` object is used for defining global variables in a project that are not explicitly imported. A great example is the `console` object. This is considered globally available in both browser and Node.js environments. In the `learn-typescript-linting` example, the code will be executed in a Node.js terminal, thus the `node` attributes is enabled. Some other common attributes include `jest`, `mocha`, `amd`, `commonjs`, and `es6`. There is no easy way to know which attributes need to be enabled; it is recommended to consult [ESLint's Specifying Environments](https://eslint.org/docs/user-guide/configuring#specifying-environments) documentation to find out what each environment attribute provides.
+
+### 2.3 Specifying ecmaVersion
+[`[toc]`](#table-of-contents)
+
+The `parserOptions.ecmaVersion` value is based on the `target` value found in the `tsconfig.json`. A `tsconfig.json` value of `{ "target": "es5" }` is equivalent to `{ "ecmaVersion": 5 }`. Use the table below for additional mappings.
+
+| tsconfig.json `target` | .eslintrc.json `ecmaVersion` |
+| ------ | ----------- |
+| `es3` | 3 |
+| `es5` | 5 |
+| `es6` or `ES2015` | 6 or 2015 |
+| `ES2016` | 7 or 2016 |
+| `ES2017` | 8 or 2017 |
+| `ES2018` | 9 or 2018 |
+| `ES2019` | 10 or 2019 |
+| `ESNext` | 10 or 2019 |
+
+Take a look at TypeScript's `--lib` [compiler options](https://www.typescriptlang.org/docs/handbook/compiler-options.html) to learn how to include unique library files in the compilation. By setting `target` to either `es5` or `es6`, TypeScript will automatically import a set of libraries (i.e. `{ target: es5 } = { lib: ['DOM', 'ES5', 'ScriptHost']}`).
+
+### 2.4 Creating ESLint npm script
+[`[toc]`](#table-of-contents)
+
+Now that ESLint is configured, create a new npm script in `package.json`:
```

Comment:
for those who aren't experienced with config files, make it more clear that you're adding it to the already existing "script"